source/vir/src/def.rs

```diff
@@ -243,81 +243,77 @@ pub fn suffix_global_id(ident: &Ident) -> Ident {
     Arc::new(ident.to_string() + SUFFIX_GLOBAL)
 }
 
-pub fn suffix_local_expr_id(ident: &Ident) -> Ident {
-    Arc::new(ident.to_string() + SUFFIX_LOCAL_EXPR)
-}
-
+// TODO: this is currently only used by debugger.rs; consider replacing this
 pub fn suffix_local_stmt_id(ident: &Ident) -> Ident {
     Arc::new(ident.to_string() + SUFFIX_LOCAL_STMT)
 }
 
-pub(crate) fn unique_bound(id: &Ident) -> UniqueIdent {
-    UniqueIdent { name: id.clone(), local: None }
+// TODO: get rid of this
+pub(crate) fn unique_bound(id: &VarIdent) -> VarIdent {
+    id.clone()
 }
 
-pub(crate) fn unique_local(id: &Ident) -> UniqueIdent {
-    UniqueIdent { name: id.clone(), local: Some(0) }
+// TODO: get rid of this
+pub(crate) fn unique_local(id: &VarIdent) -> VarIdent {
+    id.clone()
 }
 
-pub fn suffix_local_unique_id(ident: &UniqueIdent) -> Ident {
-    let UniqueIdent { name: x, local: id } = ident;
-    match id {
-        None => suffix_local_expr_id(x),
-        Some(0) => suffix_local_stmt_id(x),
-        Some(i) => {
-            Arc::new(format!("{}{}{}{}", x.to_string(), SUFFIX_LOCAL_EXPR, i, SUFFIX_LOCAL_STMT))
-        }
-    }
+// TODO: get rid of this
+pub fn suffix_local_unique_id(ident: &VarIdent) -> Ident {
+    use crate::ast_util::LowerUniqueVar;
+    ident.lower()
 }
 
-pub fn rm_suffix_local_id(ident: &Ident) -> Ident {
-    let mut name = ident.to_string();
-    if name.ends_with(SUFFIX_LOCAL_STMT) {
-        name = name[..name.len() - SUFFIX_LOCAL_STMT.len()].to_string();
-    }
-    match name.rfind(SUFFIX_LOCAL_EXPR) {
-        None => {}
-        Some(i) => {
-            name = name[..i].to_string();
-        }
-    }
-    Arc::new(name)
+pub fn subst_rename_ident(x: &VarIdent, n: u64) -> VarIdent {
+    // In principle, we could get renamed more than once, so accumulate n as a suffix
+    x.push_suffix(&format!("{SUBST_RENAME_SEPARATOR}{n}"))
```

Comment:
I don't think this is necessary. `subst_rename_binders` already checks that the new subst index won't trample any existing 'free_vars'. And I don't see why nesting the idents would help anyway.